sunxspex/emission.py

```diff
@@ -0,0 +1,814 @@
+    points evaluated until convergence, specified by the input rerr, is obtained, or the maximum
+    number of points, specified by the input maxfcn, is reached. If integral convergence is not
+    achieved, this function raises a ValueError when either the maximum number of function
+    evaluations is performed or the number of Gaussian points to be evaluated exceeds maxfcn.
+    Maxfcn should be less than or equal to 2^nlim, or 4096 with nlim = 12. This function splits the
+    numerical integration into up to three parts and returns the sum of the parts. This avoids
+    numerical problems with discontinuities in the electron distribution function at eelow and
+    eebrk.
+
+    Parameters
+    ----------
+    model : `str`
+        Electron model either `thick-target` or `thin-target`
+    photon_energies : `numpy.array`
+        Array containing lower integration limits
+    maxfcn : `int`
+        Maximum number of points used in Gaussian quadrature integration
+    rerr : `float`
+        Desired relative error for integral evaluation
+    eelow : `float`
+        Low energy electron cutoff
+    eebrk : `float`
+        Break energy
+    eehigh : `float`
+        High energy electron cutoff
+    p : `float`
+        Slope below the break energy
+    q : `float`
+        Slope above the break energy
+    z : `float`
+        Mean atomic number of plasma
+    efd : `bool`
+        True - electron flux density distribution, False - electron density distribution. This
+        input is not used in the main routine, but is passed to Brm_Fthin()
+
+    Returns
+    -------
+    `tuple`
+        (DmlinO, irer) Array of integral evaluation and array of error flags
+
+    References
+    ----------
+    Initial version modified from SSW
+    `Brm2_DmlinO <https://hesperia.gsfc.nasa.gov/ssw/packages/xray/idl/brm2/brm2_dmlino.pro>`_ and
+    `Brm2_Dmlin <https://hesperia.gsfc.nasa.gov/ssw/packages/xray/idl/brm2/brm2_dmlin.pro>`_.
+
+    """
+    mc2 = const.get_constant('mc2')
+    clight = const.get_constant('clight')
+
+    if not eelow <= eebrk <= eehigh:
+        raise ValueError(f'Condition eelow <= eebrek <= eehigh not satisfied '
+                         f'({eelow}<={eebrk}<={eehigh}).')
+
+    # Create arrays for integral sums and error flags.
+    intsum1 = np.zeros_like(photon_energies, dtype=np.float64)
+    ier1 = np.zeros_like(photon_energies, dtype=np.float64)
+    intsum2 = np.zeros_like(photon_energies, dtype=np.float64)
+    ier2 = np.zeros_like(photon_energies, dtype=np.float64)
+    intsum3 = np.zeros_like(photon_energies, dtype=np.float64)
+    ier3 = np.zeros_like(photon_energies, dtype=np.float64)
+
+    P1 = np.where(photon_energies < eelow)[0]
+    P2 = np.where(photon_energies < eebrk)[0]
+    P3 = np.where(photon_energies <= eehigh)[0]
+
+    # Part 1, below en_val[0] (usually eelow)
+    if model == 'thick-target':
+        if P1.size > 0:
+            print('Part1')
+            a_lg = np.log10(photon_energies[P1])
```

Comment:
Are you only interested on the first element/dimension of P1? (as taken from line 574)

Reply:
It's 1-d so could probably just be `if P1:`